### PR TITLE
ipv4 secrets

### PR DESCRIPTION
**Pull request by ben851**
# Summary | Résumé

ipv4 geolocate secrets managed by terraform

## Related Issues | Cartes liées

* https://app.zenhub.com/workspaces/notify-planning-core-6411dfb7c95fb80014e0cab0/issues/gh/cds-snc/notification-planning-core/299

## Before merging this PR

Read code suggestions left by the
[cds-ai-codereviewer](https://github.com/cds-snc/cds-ai-codereviewer/) bot. Address
valid suggestions and shortly write down reasons to not address others. To help
with the classification of the comments, please use these reactions on each of the
comments made by the AI review:

| Classification      | Reaction | Emoticon |
|---------------------|----------|----------|
| Useful              | +1       | 👍        |
| Noisy               | eyes     | 👀        |
| Hallucination       | confused | 😕        |
| Wrong but teachable | rocket   | 🚀        |
| Wrong and incorrect | -1       | 👎        |

The classifications will be extracted and summarized into an analysis of how helpful
or not the AI code review really is.

## Test instructions | Instructions pour tester la modification

TF Apply works

## Release Instructions | Instructions pour le déploiement

None.

## Reviewer checklist | Liste de vérification du réviseur

* [x] This PR does not break existing functionality.
* [x] This PR does not violate GCNotify's privacy policies.
* [x] This PR does not raise new security concerns. Refer to our GC Notify Risk Register document on our Google drive.
* [x] This PR does not significantly alter performance.
* [x] Additional required documentation resulting of these changes is covered (such as the README, setup instructions, a related ADR or the technical documentation).

> ⚠ If boxes cannot be checked off before merging the PR, they should be moved to the "Release Instructions" section with appropriate steps required to verify before release. For example, changes to celery code may require tests on staging to verify that performance has not been affected.

### CODE REVIEW

File: env/variables.tf [hunk 1]
```diff
@@ -1060,3 +1060,8 @@ variable "admin_a11y_tracker_key" {
   default   = "prodonly"
 }
 
+variable "ipv4_maxmind_license_key" {
+  type      = string
+  sensitive = true
+  default   = "prodonly"
+}
```
[the remark this answers is not in the document]
Comment (ben851) [reply env/variables.tf:1067]: This is a placeholder

File: aws/github/ipv4-geolocate-secrets.tf [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  count           = var.env == "production" ? 1 : 0
+  repository      = data.github_repository.ipv4_geolocate.name
+  secret_name     = "MAXMIND_LICENSE_KEY"
+  plaintext_value = var.ipv4_maxmind_license_key
+}
+
+resource "github_actions_secret" "ipv4_op_service_account_token" {
+  count           = var.env == "production" || var.env == "staging" ? 1 : 0
+  repository      = data.github_repository.ipv4_geolocate.name
+  secret_name     = "${upper(var.env)}_OP_SERVICE_ACCOUNT_TOKEN"
```
[the remark this answers is not in the document]
Comment (ben851) [reply aws/github/ipv4-geolocate-secrets.tf:11]: No go away

File: aws/github/respositories.tf [hunk 3]
```diff
@@ -20,4 +20,8 @@ data "github_repository" "notification_documentation" {
 
 data "github_repository" "notification_document_download" {
   name = "notification-document-download-api"
+}
+
+data "github_repository" "ipv4_geolocate" {
+  name = "ipv4-geolocate-webservice"
 }
```
[the remark this answers is not in the document]
Comment (ben851) [reply aws/github/respositories.tf]: I added a newline

File: aws/github/ipv4-geolocate-secrets.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (ben851) [reply aws/github/ipv4-geolocate-secrets.tf:11]: no go away

File: env/variables.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (ben851) [reply env/variables.tf:1067]: This is a better PR review but unfortunately we need this to stop TG from complaining

**Review by sastels — APPROVED** [review]

**Pull request merged**